src/dispatching/dialogue/storage/mod.rs

```diff
@@ -74,64 +74,59 @@ pub trait Storage<D> {
     {
-        struct Eraser<S>(Arc<S>);
-
-        impl<D, S> Storage<D> for Eraser<S>
-        where
-            S: Storage<D> + Send + Sync + 'static,
-            S::Error: std::error::Error + 'static,
-        {
-            type Error = Box<dyn std::error::Error>;
-
-            fn remove_dialogue(
-                self: Arc<Self>,
-                chat_id: i64,
-            ) -> BoxFuture<'static, Result<(), Self::Error>>
-            where
-                D: Send + 'static,
-            {
-                Box::pin(async move {
-                    Arc::clone(&self.0).remove_dialogue(chat_id).await.map_err(|e| e.into())
-                })
-            }
-
-            fn update_dialogue(
-                self: Arc<Self>,
-                chat_id: i64,
-                dialogue: D,
-            ) -> BoxFuture<'static, Result<(), Self::Error>>
-            where
-                D: Send + 'static,
-            {
-                Box::pin(async move {
-                    Arc::clone(&self.0)
-                        .update_dialogue(chat_id, dialogue)
-                        .await
-                        .map_err(|e| e.into())
-                })
-            }
-
-            fn get_dialogue(
-                self: Arc<Self>,
-                chat_id: i64,
-            ) -> BoxFuture<'static, Result<Option<D>, Self::Error>> {
-                Box::pin(async move {
-                    Arc::clone(&self.0).get_dialogue(chat_id).await.map_err(|e| e.into())
-                })
-            }
-        }
-
         Arc::new(Eraser(self))
     }
 }
 
-/// A storage with an erased error type.
-pub type ErasedStorage<D> = Arc<dyn Storage<D, Error = Box<dyn std::error::Error>>>;
+struct Eraser<S>(Arc<S>);
+
+impl<D, S> Storage<D> for Eraser<S>
+where
+    S: Storage<D> + Send + Sync + 'static,
+    S::Error: std::error::Error + Send + Sync + 'static,
+{
+    type Error = Box<dyn std::error::Error + Send + Sync>;
+
+    fn remove_dialogue(self: Arc<Self>, chat_id: i64) -> BoxFuture<'static, Result<(), Self::Error>>
+    where
+        D: Send + 'static,
+    {
+        Box::pin(
+            async move { Arc::clone(&self.0).remove_dialogue(chat_id).await.map_err(|e| e.into()) },
+        )
+    }
+
+    fn update_dialogue(
+        self: Arc<Self>,
+        chat_id: i64,
+        dialogue: D,
+    ) -> BoxFuture<'static, Result<(), Self::Error>>
+    where
+        D: Send + 'static,
+    {
+        Box::pin(async move {
+            Arc::clone(&self.0).update_dialogue(chat_id, dialogue).await.map_err(|e| e.into())
+        })
+    }
+
+    fn get_dialogue(
+        self: Arc<Self>,
+        chat_id: i64,
+    ) -> BoxFuture<'static, Result<Option<D>, Self::Error>> {
+        Box::pin(
+            async move { Arc::clone(&self.0).get_dialogue(chat_id).await.map_err(|e| e.into()) },
+        )
+    }
+}
+
+pub type ErasedStorage<D> =
+    dyn Storage<D, Error = Box<dyn std::error::Error + Send + Sync>> + Send + Sync;
```

Comment:
Can you move this up, just before the trait?

Reply:
Fixed, also added documentation (missed it somehow).